src/components/NotesList.jsx

```diff
@@ -7,7 +7,7 @@ import "./NotesList.css";
 function NotesList(props) {
   function renderNotesList(notes) {
     return [{}].concat(notes).map((note, i) => i !== 0
-      ? <NotesListItem key={note.noteId} note={note}></NotesListItem>
+      ? <NotesListItem key={i} note={note}></NotesListItem>
```

Comment:
I would let noteId as key since index can lead to performance and component state inconsitency if the order of items may change.

Alternatively you can go as:

  function renderNotesList(notes) {
    return <>
      <NotesListItemCreate key="new" ></NotesListItemCreate>
      {notes.map((note) => <NotesListItem key={note.noteId} note={note}></NotesListItem>)}
    </>;
  }